### PR TITLE
Adding a new parameter to support custom plugin location

### PR DESCRIPTION
**Pull request by seanschoi**
We are investigating a more flawless compilation process for p4 to ovs compiler.
In order to do so, we would like the ability to customize the location of the plugins to use for the template.
I believe this is necessary and helpful for all members of the community to add this.

### CODE REVIEW

File: p4c_bm/smart.py [hunk 1]
```diff
@@ -1222,8 +1227,12 @@ def render_all_files(render_dict, gen_dir, with_thrift = False, with_plugin_list
                             prefix = gl.tenjin_prefix)
     if len(with_plugin_list) > 0:
         for s in with_plugin_list:
-            plugin_dir =  plugin_base + s
-            plugin_files = gen_file_lists(plugin_dir, gen_dir+'/plugin/'+s, render_dict["public_inc_path"])
+            if with_plugin_path:
+                plugin_dir = os.path.join(with_plugin_path, '') + s
```
Comment (antoninbas) [inline p4c_bm/smart.py]: what about `os.path.join(with_plugin_path, s)`?

File: p4c_bm/smart.py [hunk 2]
```diff
@@ -1222,8 +1227,12 @@ def render_all_files(render_dict, gen_dir, with_thrift = False, with_plugin_list
...
+            else:
+                plugin_dir =  plugin_base + s
+            plugin_files = gen_file_lists(plugin_dir, gen_dir+'/plugin/'+s,
```
Comment (antoninbas) [inline p4c_bm/smart.py]: can this be replaced with `os.path.join(gen_dir, 'plugin', s)`?

File: p4c_bm/smart.py [hunk 3]
```diff
@@ -1222,8 +1227,12 @@ def render_all_files(render_dict, gen_dir, with_thrift = False, with_plugin_list
                             prefix = gl.tenjin_prefix)
     if len(with_plugin_list) > 0:
         for s in with_plugin_list:
-            plugin_dir =  plugin_base + s
-            plugin_files = gen_file_lists(plugin_dir, gen_dir+'/plugin/'+s, render_dict["public_inc_path"])
+            if with_plugin_path:
+                plugin_dir = os.path.join(with_plugin_path, '') + s
+            else:
+                plugin_dir =  plugin_base + s
```
Comment (antoninbas) [inline p4c_bm/smart.py]: `os.path.join(plugin_base, s)`?

File: p4c_bm/smart.py [hunk 4]
```diff
@@ -314,8 +314,10 @@ def render_dict_populate_fields(render_dict, hlir, meta_config_json):
     render_dict["pre_metadata_name_map"] = pre_metadata_name_map
 
     render_dict["extra_metadata_name_map"] = {}
+    extra_metadata_name_map = {}
```
Comment (antoninbas) [inline p4c_bm/smart.py:317]: how does this relate to the rest of the pull request?

**Comment by antoninbas** [comment]
Hi Sean,
Thanks for your Pull Request. Before I can accept it, I need to make sure that you or your organization has signed the Apache CLA for p4.org?
Also, most of the development effort has shifted from `p4c-behavioral` to [bmv2](https://github.com/p4lang/behavioral-model), and there are plans to deprecate this repo soon.

**Comment by seanschoi** [comment]
Hello Antonin
Thank you for your comments. I am part of Stanford and a student of Prof McKeown, so I think we probably have already signed the Apache CLA, but let me also get a final confirmation.
As far as other changes, I have made the changes.

Thank you

**Pull request merged**